contracts/TimeMedianDataFeedOracle.sol

```diff
@@ -0,0 +1,137 @@
+pragma solidity ^0.4.24;
+
+import "tidbit/contracts/DataFeedOracles/DataFeedOracleBase.sol";
+import "@aragon/os/contracts/lib/math/SafeMath.sol";
+
+contract TimeMedianDataFeedOracle is DataFeedOracleBase {
+  using SafeMath for uint;
+
+  event MedianizedByTimeframe(bytes32 median, uint startDate, uint endDate);
+  event MedianizedByOrderedDates(bytes32 median, uint[] dates);
+
+  mapping(uint => bool) dateAlreadyAccountedFor; // transitory data structure useful only during function call medianizeByDates
+
+  function medianizeByTimeframe(uint startDate, uint endDate)
```

Comment:
gonna nitpick formatting .. if it's multiline, the keywords should be indented, and the open bracket should be on a newline ... see https://github.com/OpenZeppelin/openzeppelin-eth/blob/64cf73d6466aacce982bf1d30d19319b835a51b9/contracts/introspection/ERC165Checker.sol#L96 as an example

Reply:
@emaG3m what if instead of startData/endDate, the tx executer passed in startIndex/endIndex? It's easy to get the index you want using the SetResult event, and passing them in would avoid having to iterate to find the dates.

Reply:
thinking maybe `function medianizeByTimeframe(int startDateIndex, int endDateIndex, uint startDate, uint endDate)` ... it can check that the indexes match the dates. And maybe passing 0 for the dates skips this check.